docs/docs/modules/typescript/index.md

```diff
@@ -0,0 +1,697 @@
+
+> Note: SpacetimeDB also provides a TypeScript **client** SDK at `spacetimedb/sdk`, as well as integrations for frameworks like `spacetimedb/react`. This guide focuses exclusively on the **server-side module** library.
+
+If you’re new to TypeScript, see the [TypeScript Handbook](https://www.typescriptlang.org/docs/handbook/intro.html). For a guided introduction to modules, see the [TypeScript Module Quickstart](/docs/module/typescript/quickstart).
+
+# Overview
+
+SpacetimeDB modules interact with the outside world via two mechanisms: **tables** and **reducers**.
+
+* [Tables](#tables) store data; public tables are queryable and subscribable by clients.
+* [Reducers](#reducers) are functions that can read and write tables and are callable over the network.
+
+Reducers are atomic and deterministic, there’s no direct filesystem or network access (e.g., `fs`, `fetch`). They execute inside the database with ACID guarantees.
+
+A minimal module looks like this:
+
+```ts
+import { schema, table, t, type RowObj } from 'spacetimedb/server';
+
+// Define a table that is publicly readable by clients
+const players = table({ name: 'players', public: true }, {
+  id: t.u32().primaryKey().autoInc(),
+  name: t.string(),
+});
+
+// Compose a schema from one or more tables
+const spacetimedb = schema(players);
+
+// Define a reducer that inserts a row
+spacetimedb.reducer('add_player', { name: t.string() }, (ctx, { name }) => {
+  ctx.db.players.insert({ id: 0, name });
+});
+```
+
+Reducers don’t return data directly; instead, clients read from tables or subscribe for live updates.
+
+Tables and reducers can use any types built with `t.*` (e.g., `t.string()`, `t.i32()`) or composite types defined with `t.object`, `t.enum`, `t.array`, or `t.option`.
+
+# Setup
+
+1. **[Install the CLI](/install)**
+
+2. **Initialize a TypeScript module project**
+
+   ```bash
+   spacetime init --lang typescript my-project
+   cd my-project
+   ```
+
+   This creates a scaffold with a sample module entrypoint and `package.json`.
+
+3. **Develop**
+
+   * Add tables with `table(...)` and reducers with `spacetimedb.reducer(...)` in your source.
+
+4. **Build and publish**
+
+   ```bash
+   spacetime login
+   spacetime publish <MY_DATABASE_NAME>
+   # Example: spacetime publish silly_demo_app
+   ```
+
+Publishing bundles your code into a JavaScript bundle, and creates a database and installs your bundle in that database. The CLI outputs the database’s **name** and **Identity** (a hex string). Save this identity for administration tasks like `spacetime logs <DATABASE_NAME>`.
+
+Re-publishing updates your module in place with [automatic migrations](#automatic-migrations) where possible:
+
+```bash
+spacetime publish <DATABASE_NAME>
```

Comment:
I might clarify here that you need to use the same name from step 1 if you want to update the database instead of creating a new one.